ui/components/app/incoming-trasaction-toggle/useIncomingTransactionToggle.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+import { TEST_CHAINS } from '../../../../shared/constants/network';
+
+export const useIncomingTransactionToggle = ({
```

Comment:
Can we move this component to `hooks` directory?

Reply:
Hey Nidhi, i found it is easier to develop and read when having hooks separately to DOM. eg: i don't have to scroll up and down to find the usage when DOM is huge.
@brad-decker helped to initialize a quick survey around this topic [here](https://consensys.slack.com/archives/GTQAGKY5V/p1690997278733539), and for now i would like to keep it outside component unless one day it becomes inconvenient ~ hope this is okay 🙏🏻

Reply:
@DDDDDanica until there is a formal acknowledgement of our patterns, lets not initiate new patterns by doing this now. Please move to hooks directory

Reply:
Thanks for chasing this up and raise some opinions. But for now you guys are correct, it is better not to introduce new patterns in an individual feature branch.  🙏🏻